### PR TITLE
Add more tests for ConfigItem and clarify error message

### PR DESCRIPTION
**Pull request by KrauseFx**

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a changelog summary in the Pull Request description @KrauseFx</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: fastlane_core/spec/config_item_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+describe FastlaneCore do
+  describe FastlaneCore::ConfigItem do
+    describe "ConfigItem input validation" do
+      it "doesn't raise an error if everything's valid" do
+        result = FastlaneCore::ConfigItem.new(key: :foo,
+                                     short_option: "-f",
+                                     description: "foo")
+        expect(result).to be_a(FastlaneCore::ConfigItem)
```
Comment (mfurtak) [inline fastlane_core/spec/config_item_spec.rb]: This doesn't seem worth testing. If we call a constructor on a class and have the result not `be_a` thing of that class, we're in trouble :smile:

**Comment by mfurtak** [comment]
Moar tests! ❤️

**Review by mfurtak — APPROVED** [review]

File: fastlane_core/spec/config_item_spec.rb (hunk 1 above)
Comment (KrauseFx) [inline fastlane_core/spec/config_item_spec.rb]: Haha good point 👍

**Comment by mfurtak** [comment]
:ship: it

**Pull request merged**